src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -16935,6 +16941,79 @@ instruct encode_ascii_array(iRegP_R2 src, iRegP_R1 dst, iRegI_R3 len,
   ins_pipe(pipe_class_memory);
 %}
 
+//----------------------------- CompressBits/ExpandBits ------------------------
+
+instruct compressBitsI_reg(iRegINoSp dst, iRegIorL2I src, iRegIorL2I mask,
+                           vRegF tdst, vRegF tsrc, vRegF tmask) %{
+  match(Set dst (CompressBits src mask));
```

Comment:
I suppose the predicate-stmt is not needed here, since the check has already been done in `match_rule_supported()` helper.

Reply:
That's a good point. x86 rules all appear to omit any checks that appear in `match_rule_supported` (in most cases they have no predicate, in others they have a predicate that includes a further sub-constraint).

For AArch64 the predicate test in `match_rule_supported` is omitted for `OP_OnSpinWait` but retained for `Op_CacheWB`, `CacheWBPreSync` and `CacheWBPostSync`. We should probably make this consistent by removing the repeat predicates for those last three cases as well.

Reply:
The CompressBits and ExpandBits nodes won't even be emitted, so the predicates would be redundant here - this is a consequence of them being intrinsics - otherwise it would be the Java implementations of the methods.
I can deal with the CacheWBs in a separate PR.

Reply:
Thanks Stuart